libs/blocks/section-metadata/section-metadata.js

```diff
@@ -34,6 +36,37 @@ function handleTopHeight(section) {
   const headerHeight = document.querySelector('header').offsetHeight;
   section.style.top = `${headerHeight}px`;
 }
+
+export function promoIntersectObserve(el, startPoint, options = {}) {
+  let abovePromoStart = false;
+  const io = new IntersectionObserver((entries, observer) => {
+    entries.forEach(async (entry) => {
+      if (el.classList.contains('close-sticky-section')) return observer.unobserve(entry.target);
+      const isPromoStart = startPoint ? entry.target === startPoint : null;
+      abovePromoStart = isPromoStart
+        ? (entry.isIntersecting || entry.boundingClientRect.y > 0)
+        : abovePromoStart;
+      if (entry.isIntersecting || (!entry.isIntersecting && isPromoStart && abovePromoStart)) el.classList.add('hide-sticky-section');
+      else if (!entry.isIntersecting && !abovePromoStart) el.classList.remove('hide-sticky-section');
+      return el;
+    });
+  }, options);
+  return io;
+}
+
+async function handleStickyPromobar(section) {
```

Comment:
To @mokimo 's earlier point, I don't see any `await` declarations, `async` is probably redundant. The `await` on L81 might also be redundant. Am I missing something? Is there some async code going on somewhere?

Reply:
@mokimo @overmyheadandbody I have made the requested changes.